Automattic/woocommerce-payments
# Fix Multi-Currency formatting on My Account > Orders page

Fixes #7409

#### Changes proposed in this Pull Request

Multi-Currency formatting for the My Account > Order page was broken, this PR aims to fix that.

* Set the `order_currency` in `FrontendCurrencies` when the `woocommerce_order_get_total` filter is used via the `maybe_init_order_currency_from_order_total_prop` method. This will do a check to see if the `order_currency` should be used via the `should_use_order_currency` method and will set the `order_currency` for use accordingly.
* Clears the `order_currency` when the `woocommerce_get_formatted_order_total` filter is used via the `maybe_clear_order_currency_after_formatted_order_total` method. This does the check via the `should_use_order_currency`, as well, and clears the `order_currency` if it is already not `null`.

#### Testing instructions

> [!NOTE]  
> This testing assumes your store's default currency is USD and that you have Multi-Currency enabled.

* Do not check out this branch yet.
* Place an order using the EUR currency.
  * If you do not have EUR enabled, this can be done under WooCommerce > Settings > Multi-Currency.
* Place an order using the default USD currency.
* Navigate on the front end to My Account > Orders.
* Your orders should have totals that resemble this formatting:
  * €14.00 EUR for 1 item
  * $15.00 USD for 1 item
* Check out this branch, then refresh the My Account > Orders page.
* Your orders should have totals that resemble this formatting:
  * 14,00 € EUR for 1 item
  * $15.00 USD for 1 item
* Only the formatting of EUR should have changed, but it may be different based on your locale.


-------------------


**Post merge**
